### PR TITLE
True italic fonts.

### LINKED ISSUE #711: True italic font support
Currently, italic fonts are achieved by making the current font oblique. It would be great if it would use the italic font instead.

Here's how I envision it could work: Similar to using a custom font with `~/.termux/font.ttf`, put an italic font in a file called `~/.termux/font-italic.ttf`. Then run this:
```bash console
$ termux-reload-settings
$ echo -e '\e[3m light fixture \e[0m' # characters that look different in italics
```
and it would print this:
_` light fixture `_

---
#### Regular:
screenshot_20180610-172126

#### Italic:
screenshot_20180610-173553

### PR DESCRIPTION
**Pull request by j0ng4b**
Closes #711. Adds support to true italic fonts. If an italic font file (`~/.termux/font-italic.ttf`) not exists, fallback to previous italic effect.

One thing that maybe can be an issue is to use two different kinds of fonts. This isn't easy to handle with my current knowledge of the code then should be avoided if possible (sane people will not use two different fonts, I hope...).

### CODE REVIEW

**Comment by 2096779623** [comment]
If this PR is merged please add a corresponding tip here:
https://wiki.termux.com/wiki/Termux:Styling

**Comment by MrAdityaAlok** [comment]
> One thing that maybe can be an issue is to use two different kinds of fonts.

I couldn't understand that. Can you provide an example?

I tried this PR with two different fonts and it works:
- Regular font (~/.termux/font.ttf): Hack regular nerd font.
- Italic font (~/.termux/font-italic.ttf): Inconsolata italic nerd font.

Screenshot_2022-07-06-10-41-16-125

**Comment by CanKolay3499** [comment]
Why is this not merged?

**Comment by 2096779623** [comment]
> Why is this not merged?

image

**Comment by rendiix** [comment]
Waiting for this project to be fixed and merged

**Comment by luisdavim** [comment]
@j0ng4b @agnostic-apollo can this be merged?

**Comment by HardcodedCat** [comment]
There's failures at building. it can't be merged yet
> Task :terminal-emulator:bundleLibCompileToJarDebug
> Task :terminal-view:generateDebugBuildConfig
> Task :terminal-view:javaPreCompileDebug
/home/runner/work/termux-monet/termux-monet/terminal-view/src/main/java/com/termux/view/TerminalRenderer.java:237: error: cannot find symbol
>            if (cursorStyle == TerminalEmulator.TERMINAL_CURSOR_STYLE_UNDERLINE) cursorHeight /= 4.;                                                                               
>  symbol:   variable cursorHeight
  location: class TerminalRenderer
/home/runner/work/termux-monet/termux-monet/terminal-view/src/main/java/com/termux/view/TerminalRenderer.java:239: error: cannot find symbol
> Task :terminal-view:compileDebugJavaWithJavac FAILED
            canvas.drawRect(left, y - cursorHeight, right, y, mTextPaint);
                                      ^
  symbol:   variable cursorHeight
  location: class TerminalRenderer
Note: /home/runner/work/termux-monet/termux-monet/terminal-view/src/main/java/com/termux/view/TerminalView.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
2 errors

**Comment by Xemptuous** [comment]
Waiting for this PR merge conflicts to be resolved to see if it fixes my italic clipping issues with neovim. Any progress?

**Comment by rendiix** [comment]
Hopefully it's fixed soon

**Comment by hacker-hackman** [comment]
Please, resolve the conflicts 🙏

**Comment by HardcodedCat** [comment]
For anyone who wants to build, the PR is still working, there's very simple conflicts

**Comment by mrtnvgr** [comment]
Any updates?

**Comment by j0ng4b** [comment]
When I get some time I'll work on improve this maybe add true bold fonts also

**Comment by Duckulus** [comment]
Hey, are there any updates on this?

**Comment by JaysonEdenfield** [comment]
i think bro forgot about this

**Comment by Alomgir-Bhuiyan** [comment]
Screenshot_20260531-094440-1


Help requested,
Termux: 0.119.0-beta.3

File: terminal-view/src/main/java/com/termux/view/TerminalView.java [hunk 1]
```diff
@@ -459,26 +446,18 @@ public void onScreenUpdated(boolean skipScrolling) {
...
         updateSize();
     }
 
-    public void setTypeface(Typeface newTypeface) {
-        mRenderer = new TerminalRenderer(mRenderer.mTextSize, newTypeface);
+    public void setTypeface(Typeface newTypeface, Typeface newItalicTypeface) {
+        mRenderer = new TerminalRenderer(mRenderer.mTextSize, newTypeface, newItalicTypeface);
         updateSize();
         invalidate();
     }
```
Comment (WALElionel) [inline terminal-view/src/main/java/com/termux/view/TerminalView.java:463]: j0ng4b:master

Comment (Alomgir-Bhuiyan) [reply terminal-view/src/main/java/com/termux/view/TerminalView.java:463]: So i have to build the whole thing with this PR?

**Pull request closed without merging**